### DISCUSSION TITLE
Merics to get the status of Microservices like query node, index node.

### DISCUSSION THREAD

**Discussion by navneetrajput06** [milvus-io/milvus#31321] [Q&A and General discussion]
Hi team,
We did not see any metrics coming in Prometheus where we can get the status of each milvus service. If they are down.  do we have any metrics to get that status.
Components like query node, index node etc.

Thanks,
Navneet

**Comment by yhmo** [comment]
All metrics are listed here:
https://milvus.io/docs/metrics_dashboard.md#Milvus-Metrics-Dashboard

In Grafana, there is a view "Query Node Num" under Qeruy Coordinator, "Data Node Num" under Data Coordinator, and "Index Node Num" under Index Coordinator. If a work node is down, it stops sending metrics, you will see the "XXX Node Num" changed. You still can see the history metrics for this node.
![Screenshot from 2024-03-18 10-09-56](https://github.com/milvus-io/milvus/assets/2282099/b0ca029d-54ae-42e2-9688-89257c6121aa)

**Reply by navneetrajput06** [reply]
@yhmo i have v2.3.3 version and did not see these metrics there. Are these default metrics comes with Milvus deployment or we need to export them by using some different procedure.?

**Reply by yhmo** [reply]
Just deploy a monitoring service by following this doc: https://milvus.io/docs/monitor.md
